src/cart/src/cart/crt_swim.c

```diff
@@ -114,6 +119,41 @@ static void crt_swim_srv_cb(crt_rpc_t *rpc_req)
 		rpc_req->cr_opc, rpc_swim_input->upds.ca_count,
 		self_id, rpc_swim_input->src);
 
+	rpc_priv = container_of(rpc_req, struct crt_rpc_priv, crp_pub);
+	delay = (hlc - rpc_priv->crp_req_hdr.cch_hlc) / NSEC_PER_MSEC;
+	csm->csm_msg_count++;
+
+	if (csm->csm_hlc) {
+		csm->csm_avg_delay = (csm->csm_avg_delay + delay) / 2;
+
+		D_TRACE_DEBUG(DB_TRACE, rpc_req,
+			      "%lu. delay: %lu ms, avg: %lu ms, max: %lu ms\n",
+			      csm->csm_msg_count, delay, csm->csm_avg_delay,
+			      max_delay);
+
+		if (delay > max_delay) {
```

Comment:
probably can define a max reasonable delay? To avoid the case that the SWIM msg be delayed by some un-expected situation like context not being progress for long time or network congest, that kind of unexpected msg delay should not cause changing normal SWIM protocol parameter. 
For example define MAX_RESONABLE_DELAY as 3S, then only the SWIM msg delay between [400mS, 3000mS] can cause SWIM protocol parameter tuning.

Reply:
This is condition only checks when expiration timeouts should be updated once. The timeouts will be tuned only is average delay will be more than max_delay (line 136), For once short network glitch nothing will be tuned.

Reply:
OK, I am thinking that an unreasonable SWIM msg probably should not be calculated in the average delay.
But it looks fine to me to land it first and refine it later when needed. Same for the other 2 points of my comments.